aiida/cmdline/commands/cmd_graph.py

```diff
@@ -37,33 +48,49 @@ def verdi_graph():
 @click.option(
     '-d',
     '--descendant-depth',
-    help='The maximum depth when recursing through the descendants, if not set it will recurse to the end',
+    help='The maximum depth when recursing through the descendants. If not set it will recurse to the end.',
     type=click.IntRange(min=0))
-@click.option('-o', '--outputs', is_flag=True, help='Always show all outputs of a calculation.')
-@click.option('-i', '--inputs', is_flag=True, help='Always show all inputs of a calculation.')
+@click.option('-o', '--process-out', is_flag=True, help='Show outgoing links for all processes.')
+@click.option('-i', '--process-in', is_flag=True, help='Show incoming links for all processes.')
+@options.VERBOSE(help="Print verbose information of the graph traversal.")
 @click.option(
-    '-f',
-    '--output-format',
-    help='The output format, something that can be recognized by graphviz'
-    '(see http://www.graphviz.org/doc/info/output.html).',
+    '-e',
+    '--engine',
+    help="The graphviz engine, e.g. 'dot', 'circo', ... "
+    "(see http://www.graphviz.org/doc/info/output.html)",
     default='dot')
+@click.option('-f', '--output-format', help="The output format used for rendering ('pdf', 'png', etc.).", default='pdf')
+@click.option('-s', '--show', is_flag=True, help="Open the rendered result with the default application.")
 @decorators.with_dbenv()
-def generate(root_node, ancestor_depth, descendant_depth, outputs, inputs, output_format):
-    """Generate a graph for a given ROOT_NODE."""
-    from aiida.tools.visualization.graphviz import draw_graph
+def generate(root_node, link_types, id_label, ancestor_depth, descendant_depth, process_out, process_in, engine,
+             verbose, output_format, show):
+    """
+    Generate a graph from a ROOT_NODE (specified by pk or uuid).
+    """
+    # pylint: disable=too-many-arguments
+    from aiida.tools.visualization import Graph
+    print_func = echo.echo_info if verbose else None
+    link_types = {"all": (), "logic": ("input_work", "return"), "data": ("input_calc", "create")}[link_types]
 
-    try:
-        exit_status, output_file_name = draw_graph(
-            root_node,
-            ancestor_depth=ancestor_depth,
-            descendant_depth=descendant_depth,
-            image_format=output_format,
-            include_calculation_inputs=inputs,
-            include_calculation_outputs=outputs)
-    except OSError as exception:
-        echo.echo_critical(str(exception))
+    echo.echo_info("Initiating graphviz engine: {}".format(engine))
```

Comment:
Maybe also use the `print_func` here, or do we always want these lines to be printed regardless of `verbose` flag?

Reply:
Well I guess they are a different level of verbosity. Your `VERBOSE` should probably employ the `count=True` functionality (see [click example](http://click.palletsprojects.com/en/5.x/options/#counting)), then `-v` would be these echos and `-vv` would also be `print_func`. I guess thats not a change for this PR, since it could affect other commands. Shall I switch to creating my own verbose option for now?

Reply:
No not necessary. If this was already there, it's indeed fine to leave it as it this